### PR TITLE
Fix collectionClass not working with the Yaml or Xml drivers.

### PR DESCRIPTION
**Pull request by trq**
The custom collections recently implemented in #1385 looks like it was never working with the Yaml (or Xml) drivers.

This PR addresses that issue.

### CODE REVIEW

File: lib/Doctrine/ODM/MongoDB/Mapping/Driver/XmlDriver.php [hunk 1]
```diff
@@ -244,11 +244,12 @@ private function addEmbedMapping(ClassMetadataInfo $class, $embed, $type)
...
         $mapping = array(
-            'type'           => $type,
-            'embedded'       => true,
-            'targetDocument' => isset($attributes['target-document']) ? (string) $attributes['target-document'] : null,
-            'name'           => (string) $attributes['field'],
-            'strategy'       => isset($attributes['strategy']) ? (string) $attributes['strategy'] : $defaultStrategy,
+            'type'            => $type,
+            'embedded'        => true,
+            'targetDocument'  => isset($attributes['target-document']) ? (string) $attributes['target-document'] : null,
+            'collectionClass' => isset($attributes['collectionClass']) ? (string) $attributes['collectionClass'] : null,
```
Comment (malarzm) [inline lib/Doctrine/ODM/MongoDB/Mapping/Driver/XmlDriver.php]: Should we use `collection-class` to be consistent?

File: lib/Doctrine/ODM/MongoDB/Mapping/Driver/XmlDriver.php [hunk 2]
```diff
@@ -291,6 +292,7 @@ private function addReferenceMapping(ClassMetadataInfo $class, $reference, $type
             'simple'           => isset($attributes['simple']) ? ('true' === (string) $attributes['simple']) : false, // deprecated
             'storeAs'          => isset($attributes['store-as']) ? (string) $attributes['store-as'] : ClassMetadataInfo::REFERENCE_STORE_AS_DB_REF_WITH_DB,
             'targetDocument'   => isset($attributes['target-document']) ? (string) $attributes['target-document'] : null,
+            'collectionClass'  => isset($attributes['collectionClass']) ? (string) $attributes['collectionClass'] : null,
```
Comment (malarzm) [inline lib/Doctrine/ODM/MongoDB/Mapping/Driver/XmlDriver.php]: Same as above

**Comment by malarzm** [comment]
@trq thanks a bunch for catching this! I have no idea why have I forgotten about them... Apart from my 2 comments above could you also please

1. Update (XSD)[https://github.com/doctrine/mongodb-odm/blob/master/doctrine-mongo-mapping.xsd] file if we agree on using hyphens
2. Add mapping tests to ensure the attribute is in fact set
3. Fix failing tests (don't bother with PHPUnit warning, I can fix them later unless you want to)

**Comment by trq** [comment]
No worries, thanks for the feedback. I'll make these changes and update shortly.

**Comment by trq** [comment]
I'm a little lost with how/where to test the mapping on the XmlDriver. Any advice much appreciated. Thanks.

File: doctrine-mongo-mapping.xsd [hunk 3]
```diff
@@ -109,6 +109,7 @@
       <xs:element name="default-discriminator-value" type="odm:default-discriminator-value" minOccurs="0"/>
     </xs:sequence>
     <xs:attribute name="target-document" type="xs:string" />
+    <xs:attribute name="collection-class" type="xs:string" />
```
Comment (malarzm) [inline doctrine-mongo-mapping.xsd]: There seem to be `collectionClass` in line 115/116 (and 168/170) on which I can't comment, could you remove that?

**Comment by malarzm** [comment]
As for XML tests you could tweak this [AbstractMappingDriverTest](https://github.com/doctrine/mongodb-odm/blob/master/tests/Doctrine/ODM/MongoDB/Tests/Mapping/AbstractMappingDriverTest.php) which is extended by each driver type we have, here's XML mapping used by the test: https://github.com/doctrine/mongodb-odm/blob/master/tests/Doctrine/ODM/MongoDB/Tests/Mapping/xml/Doctrine.ODM.MongoDB.Tests.Mapping.AbstractMappingDriverUser.dcm.xml and probably you'll need to update tested [YAML](https://github.com/doctrine/mongodb-odm/blob/master/tests/Doctrine/ODM/MongoDB/Tests/Mapping/yaml/Doctrine.ODM.MongoDB.Tests.Mapping.AbstractMappingDriverUser.dcm.yml) and annotation one too :)

**Comment by trq** [comment]
I've added some more tests, hopefully not making too much mess.

I still have one failing test that I'll look at, but I would appreciate some help if anyone want's to take a look.

File: tests/Doctrine/ODM/MongoDB/Tests/Mapping/YamlMappingDriverTest.php [hunk 4]
```diff
@@ -31,12 +31,23 @@ public function testAlternateRelationshipMappingSyntaxShouldSetDefaults()
...
-            foreach (array('strategy', 'targetDocument') as $key) {
+            foreach (array('strategy', 'targetDocument', 'collectionClass') as $key) {
                 $this->assertArrayHasKey($key, $class->fieldMappings[$embeddedField]);
             }
         }
     }
 
+    public function testGetAssociationCollectionClass()
+    {
+        $className = __NAMESPACE__.'\AbstractMappingDriverAlternateUser';
```
Comment (malarzm) [inline tests/Doctrine/ODM/MongoDB/Tests/Mapping/YamlMappingDriverTest.php]: I think you wanted to use `AbstractMappingDriverUser` as that's the class you changed mapping for

Comment (malarzm) [inline tests/Doctrine/ODM/MongoDB/Tests/Mapping/YamlMappingDriverTest.php]: P.S. sorry for such long break :)

**Comment by malarzm** [comment]
@trq thanks for making tests green again! I should start merging stuff on Monday/Tuesday as I'm off for the weekend :)

**Comment by trq** [comment]
No worries, thanks for your help.

Do you want me to squash this PR into a single commit or is there anything
else that it needs before merging ?

**Comment by malarzm** [comment]
I can squash commits and merge manually, no worries :)

**Comment by malarzm** [comment]
Merged manually in https://github.com/doctrine/mongodb-odm/commit/0a9caac5019409c0dc6d5411adbb61665db8f0a0 - thanks @trq!

**Pull request closed without merging**